src/test/assembly/aarch64-pointer-auth.rs

```diff
@@ -0,0 +1,22 @@
+// Test that PAC instructions are emitted when branch-protection is specified.
+
+// min-llvm-version: 10.0.1
+// assembly-output: emit-asm
+// compile-flags: --target aarch64-unknown-linux-gnu
+// compile-flags: -C branch-protection=pac-ret+leaf
+// needs-llvm-components: aarch64
+
+#![feature(no_core, lang_items)]
+#![no_std]
+#![no_core]
+#![crate_type = "lib"]
+
+#[lang = "sized"]
+trait Sized {}
+
+// CHECK: hint #25
```

Comment:
This is an ISA artefact; some pointer authentication instructions are encoded as hints, so they are no-ops on hardware that doesn't support them. Specifically, `hint #25` is a `paciasp` instruction on hardware that supports pointer authentication (Armv8.3).

As such, this won't change, and we do need to check for this specific hint, but an updated LLVM could conceivably start disassembling it as `paciasp`. (I've not checked recently, but I think it would only do that today if Armv8.3 is specified.)

Reply:
We could certainly add `CHECK-LABEL`s though.